database/src/desktop/connection/persistent_connection.cc

```diff
@@ -135,10 +135,10 @@ PersistentConnection::~PersistentConnection() {
   // Clear OnCompletion function for pending token future
   {
     MutexLock future_lock(pending_token_future_mutex_);
-    if (pending_token_future_.status() != kFutureStatusInvalid) {
-      pending_token_future_.OnCompletion(nullptr, nullptr);
-      pending_token_future_ = Future<std::string>();
-    }
+    pending_auth_token_future_ = Future<std::string>();
```

Comment:
What it was doing before was setting the OnCompletion callback to a nullptr, which doesn't seem to be supported by the future logic.  If the future was in the completed state, it would try to callback on the nullptr here, which would crash it.  The reason it worked before was the usual OnCompletion callback was clearing the pending_token_future_, so when it was deleted this logic just wouldn't run.

Reassigning the future with ... = Future() does clear the old future out, so that should be enough.